source/opt/block_merge_util.cpp

```diff
@@ -49,14 +49,21 @@ bool IsMerge(IRContext* context, BasicBlock* block) {
   return IsMerge(context, block->id());
 }
 
-// Returns true if block |id| starts with an OpPhi instruction.
-bool StartsWithOpPhi(IRContext* context, uint32_t id) {
-  bool result = false;
-  context->cfg()->block(id)->ForEachPhiInst([&result](Instruction* /*unused*/) {
-    // If any OpPhi is found then the block must start with OpPhi.
-    result = true;
+// Requires that any OpPhi in |successor| has |predecessor| as its sole parent.
+// Replaces all uses of such OpPhis with the id associated with |predecessor|,
+// and removes the OpPhi instructions.
+void EliminateOpPhiInstructions(IRContext* context, BasicBlock* successor,
+                                BasicBlock* predecessor) {
```

Comment:
You do not actually need `predecessor` in this function.  It is only used as a check to make sure that the OpPhi instruction was a valid OpPhi.  I would suggest removing it.  We check that there number of in operands is 2 and just assume that the only predecessor was the correct one.

If not, it would be invalid spirv and it does not matter what we generate.